src/components/SIngleArticle.jsx

```diff
@@ -1,25 +1,80 @@
-import axios from "axios"
-import { useEffect, useState } from "react"
-import { useParams } from "react-router-dom"
+import axios, { Axios } from "axios";
+import { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { Link } from "react-router-dom";
 
-function SingleArticle(){
-    let {article_id} = useParams()
-    const [article,setArticle] = useState({})
+function SingleArticle() {
+  let { article_id } = useParams();
+  const [article, setArticle] = useState({});
+  const [comments, setComments] = useState([]);
 
-    useEffect(() => {
-        axios.get(`https://ncn-network.onrender.com/api/articles/${article_id}`)
-        .then((response) => {
-            console.log(response.data)
-            setArticle(response.data)
-        })
-    },[])
-console.log(article,'outside the effect')
+  useEffect(() => {
+    axios
+      .get(`https://ncn-network.onrender.com/api/articles/${article_id}`)
+      .then((response) => {
+        setArticle(response.data);
+      });
+  }, [article_id]);
 
-return(
-    <div className="singleArticle">
+  useEffect(() => {
+    axios
+      .get(
+        `https://ncn-network.onrender.com/api/articles/${article_id}/comments`
+      )
+      .then((response) => {
+        setComments(response.data);
+      });
+  }, [article_id]);
+
+  function increaseVotes(comment) {
+    const updatedComments = comments.map((singleComment) => {
+      if (singleComment.comment_id === comment.comment_id) {
+        return { ...singleComment, votes: singleComment.votes + 1 };
+      }
+      return singleComment;
+    });
+    setComments(updatedComments);
+  }
+  function decreaseVotes(comment) {
+    const updatedComments = comments.map((singleComment) => {
+      if (singleComment.comment_id === comment.comment_id) {
+        return { ...singleComment, votes: singleComment.votes - 1 };
+      }
+      return singleComment;
+    });
+    setComments(updatedComments);
+  }
```

Comment:
This is good optimistic rendering but we need the votes to persist. You'll need to make api patch requests here too. You can undo your optimistic vote adding in the catch block :)